File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def _prev(time):
    if time < 10:
        return 0
    return time - 10

def _next(time, end_time):
    if end_time - time < 10:
        return end_time
    return time + 10

def _skip(time, op_start, op_end):
    if (op_start <= time <= op_end):
        return op_end
    return time

def min_to_sec(time):
    minute, second = map(int, time.split(":"))
    return minute * 60 + second

def sec_to_min(time):
    minute = str(time // 60).zfill(2)
    second = str(time % 60).zfill(2)
    return f"{minute}:{second}"

def solution(video_len, pos, op_start, op_end, commands):
    video_len = min_to_sec(video_len)
    pos = min_to_sec(pos)
    op_start = min_to_sec(op_start)
    op_end = min_to_sec(op_end)
    pos = _skip(pos, op_start, op_end)
    
    for command in commands:
        if command == "next":
            pos = _next(pos, video_len)
        elif command == "prev":
            pos = _prev(pos)
            
        pos = _skip(pos, op_start, op_end)
            
    return sec_to_min(pos)
            

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.js
const timeToNumber = (time) => {
    const [m, s] = time.split(":").map(Number);
    return m * 60 + s;
}

const timeToString = (time) => {
    const m = parseInt(time / 60, 10).toString().padStart(2, '0');
    const s = (time % 60).toString().padStart(2, '0');
    return `${m}:${s}`;
}

const execPrev = (time) => {
    return time < 10 ? 0 : time - 10;
}

const execNext = (time, endTime) => {
    return endTime - time < 10 ? endTime : time + 10;
}

const execSkip = (time, opStart, opEnd) => {
    if (opStart <= time && time <= opEnd) {
        return opEnd
    }
    return time
}

function solution(video_len, pos, op_start, op_end, commands) {
    let now = timeToNumber(pos);

    const endTime = timeToNumber(video_len);
    const opStart = timeToNumber(op_start);
    const opEnd = timeToNumber(op_end);
    now = execSkip(now, opStart, opEnd);
    
    for (const command of commands) {
        if (command === "next") {
            now = execNext(now, endTime);
        } else if (command === "prev") {
            now = execPrev(now);
        }
        now = execSkip(now, opStart, opEnd);
    }
    return timeToString(now);
}